nixos/modules/services/networking/wakeonwlan.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.wakeonwlan;
+
+  mkWowlanService = name: cfg:
+    nameValuePair "wowlan-${name}" {
+      description = "Enable WoWLAN for interface ${name}";
+      requires = [ "network.target" ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+      };
+      script = ''
+        ${pkgs.iw}/bin/iw ${name} wowlan enable ${concatStringsSep " " cfg.methods}
+      '';
+    };
+in
+{
+  options.services.wakeonwlan.interfaces = mkOption {
+    default = { };
+    description = "Wireless interfaces where you want to enable WoWLAN";
+    example = literalExample ''
+      {
+        phy0.methods = [
+          "magic-packet"
+          "disconnect"
+          "gtk-rekey-failure"
+          "eap-identity-request"
+          "rfkill-release"
+        ];
+        phy2.methods = [ "any" ];
+      }
+    '';
+    type = types.attrsOf (
+      types.submodule {
+        options = {
+          methods = mkOption {
+            type = types.listOf (types.enum [
+              "4way-handshake"
+              "any"
+              "disconnect"
+              "eap-identity-request"
+              "gtk-rekey-failure"
+              "magic-packet"
+              "rfkill-release"
+            ]);
+            description = "Wake-On-WiFiLan methods for this interface.";
```

Comment:
Note that `iw` supports more methods, I've only added methods that do not need additional parameters :

```
phy <phyname> wowlan enable [any] [disconnect] [magic-packet] [gtk-rekey-failure] [eap-identity-request] [4way-handshake] [rfkill-release] [net-detect [interval <in_msecs> | scan_plans [<interval_secs:iterations>*] <interval_secs>] [delay <in_secs>] [freqs <freq>+] [matches [ssid <ssid>]+]] [active [ssid <ssid>]+|passive] [randomise[=<addr>/<mask>]]] [tcp <config-file>] [patterns [offset1+]<pattern1> ...]
		Enable WoWLAN with the given triggers.
		Each pattern is given as a bytestring with '-' in places where any byte
		may be present, e.g. 00:11:22:-:44 will match 00:11:22:33:44 and
		00:11:22:33:ff:44 etc.
		Offset and pattern should be separated by '+', e.g. 18+43:34:00:12 will match '43:34:00:12' after 18 bytes of offset in Rx packet.
		
		The TCP configuration file contains:
		  source=ip[:port]
		  dest=ip:port@mac
		  data=<hex data packet>
		  data.interval=seconds
		  [wake=<hex packet with masked out bytes indicated by '-'>]
		  [data.seq=len,offset[,start]]
		  [data.tok=len,offset,<token stream>]
		
		Net-detect configuration example:
		 iw phy0 wowlan enable net-detect interval 5000 delay 30 freqs 2412 2422 matches ssid foo ssid bar
```